microceph/ceph/service_placement_nfs.go

```diff
@@ -0,0 +1,101 @@
+package ceph
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/canonical/microceph/microceph/database"
+	"github.com/canonical/microceph/microceph/interfaces"
+)
+
+type NFSServicePlacement struct {
+	ClusterID    string `json:"cluster_id"`
+	V4MinVersion uint   `json:"v4_min_version"`
+	BindAddress  string `json:"bind_address"`
+	BindPort     uint   `json:"bind_port"`
+}
+
+func (nfs *NFSServicePlacement) PopulateParams(s interfaces.StateInterface, payload string) error {
+	err := json.Unmarshal([]byte(payload), &nfs)
+	if err != nil {
+		return err
+	}
+
+	if len(nfs.ClusterID) == 0 {
+		return fmt.Errorf("expected cluster_id to be non-empty")
+	}
```

Comment:
While the ceph docs say ClusterID can be an arbitrary string, I believe the clusterid ends up in pool and user names, also it gets rendered into the config file and cmdlines, so there's some risk here (ie. the bobby tables effect). 

I would suggest to constrain which chars to accept e.g. the POSIX portable filename set, ensure it doesn't start with a period `.` as that is for internal pools, and constrain length to something sensible like 63 chars